lib/kubernetes-deploy/runner_task.rb

```diff
@@ -15,43 +15,115 @@ def initialize(task_template, namespace, context)
       end
     end
 
-    def initialize(namespace:, context:, logger:)
+    def initialize(namespace:, context:, logger:, max_watch_seconds: nil)
       @logger = logger
       @namespace = namespace
       @kubeclient = build_v1_kubeclient(context)
       @context = context
+      @max_watch_seconds = max_watch_seconds
     end
 
     def run(*args)
       run!(*args)
       true
+    rescue DeploymentTimeoutError
+      false
     rescue FatalDeploymentError => error
-      @logger.fatal "#{error.class}: #{error.message}"
+      if error.message != error.class.to_s
+        @logger.summary.add_action(error.message)
+        @logger.print_summary(:failure)
+      end
       false
     end
 
-    def run!(task_template:, entrypoint:, args:, env_vars: [])
+    def run!(task_template:, entrypoint:, args:, env_vars: [], verify_result: true)
```

Comment:
This method is pretty long now and does things on different levels of abstractions. Can you clean this up?